var/spack/repos/builtin/packages/acts-algebra-plugins/package.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class ActsAlgebraPlugins(CMakePackage):
+    """A portable linear algebra library with multiple backends that is part of
+    the Acts ecosystem."""
+
+    homepage = "https://github.com/acts-project/algebra-plugins"
+    url = "https://github.com/acts-project/algebra-plugins/archive/refs/tags/v0.24.0.tar.gz"
+
+    maintainers("stephenswat", "wdconinc")
+
+    license("MPL-2.0", checked_by="stephenswat")
+
+    version("0.24.0", sha256="f44753e62b1ba29c28ab86b282ab67ac6028a0f9fe41e599b7fc6fc50b586b62")
+
+    variant(
+        "cxxstd",
+        default="17",
+        values=("17", "20"),
+        multi=False,
+        description="C++ standard used",
+    )
+    variant("eigen", default=False, description="Enables the Eigen plugin")
+    variant("smatrix", default=False, description="Enables the SMatrix plugin")
+    variant("vecmem", default=False, description="Enables the vecmem plugin")
+    variant("vc", default=False, description="Enables the Vc plugin")
+    variant("fastor", default=False, description="Enables the Fastor plugin")
+
+    depends_on("cmake@3.14:", type="build")
+    depends_on("vecmem@1.5.0:", when="+vecmem")
+    depends_on("eigen@3.4.0:", when="+eigen")
+    depends_on("vc@1.4.3:", when="+vc")
+    depends_on("root@6.18.0:", when="+smatrix")
+    depends_on("fastor@0.6.4:", when="+fastor")
+
+    with when("+smatrix"):
+        depends_on("root cxxstd=17", when="cxxstd=17")
+        depends_on("root cxxstd=20", when="cxxstd=20")
```

Comment:
I made one more small change: I made this package depend on a root version with the same C++ standard, as ROOT does not like to be used with a different C++ standard than the one with which it was compiled. Similarly, I removed the C++23 support in this package, as ROOT does not support it.

Reply:
You can keep `23` in the supported `cxxstd` set, and even keep the `root cxxstd=23` dependency. As long as there is no root version that supports it, it won't get picked. As soon as there is a root version, it will become possible to pick it up as a solution. And it will immediately be possible to use `23` without root.

Reply:
I tried this at first, but interestingly Spack didn't like that:

```bash
$ spack install "acts-algebra-plugins@0.24.0+vecmem+smatrix cxxstd=20"
==> Error: invalid values for variant "cxxstd" in package "root": ['23']
```

So it seems that it is evaluating the feasibility of options, even if they are not selected?

Reply:
Ah, hmmm... Maybe it needs at least one root version to support it. In that case it probably still makes sense to list it here but add a conflict for `cxxstd=23 +smatrix`.

Reply:
Good idea, I've gone with that strategy.